StoreResourceLoader.js

```diff
@@ -0,0 +1,50 @@
+const clownface = require('clownface')
+const ns = require('@tpluscode/rdf-ns-builders')
+const { fromStream } = require('rdf-dataset-ext')
+const rdf = { ...require('@rdfjs/data-model'), ...require('@rdfjs/dataset') }
+const TermSet = require('@rdfjs/term-set')
+
+class StoreResourceLoader {
+  constructor ({ store }) {
+    this.store = store
+  }
+
+  async load (term) {
+    const dataset = await fromStream(rdf.dataset(), this.store.match(null, null, null, term))
+
+    if (dataset.size === 0) {
+      return null
+    }
+
+    const types = new TermSet(clownface({ dataset, term }).out(ns.rdf.type).terms)
+
+    return {
+      term,
+      dataset,
+      types
+    }
+  }
+
+  async byClass (term) {
```

Comment:
Not sure about the method names. This one has nothing to do with class does it?

Maybe

```suggestion
  async bySubject (term) {
```

Arguably, it's actually looking up the named graph but I'd say it's an implementation detail. Otherwise could be implemented as `DESCRIBE ${term}` which justifies the "subject"

Reply:
The method name should cover what it is used for and not the logic of possible implementations. The method is always used to load the resource for a class operation. What about `forClassOperation`?

Reply:
I see. This covers how graph is used to implement the default lookup and not actually subject.

Ok, let's do `forClassOperation`/`forPropertyOperation`